pkg/kubelet/network/dns/dns_test.go

```diff
@@ -62,6 +62,23 @@ func newTestConfigurer(recorder *record.FakeRecorder, nodeRef *v1.ObjectReferenc
 		},
 		Status: arktosv1.NetworkStatus{
 			DNSServiceIP: dnsVIPOfTestNetwork,
+			Phase:        "Ready",
+		},
+	})
+	return NewConfigurer(recorder, nodeRef, nil, clusterDNS, testClusterDNSDomain, resolverConfig, networkClient.ArktosV1())
+}
+
+func newTestConfigurerWithNotReadyNetwork(recorder *record.FakeRecorder, nodeRef *v1.ObjectReference, clusterDNS []net.IP, testClusterDNSDomain, resolverConfig string) *Configurer {
```

Comment:
Does it work for non-system tenant? If yes, we should have a test case for it.

Reply:
tenancy is orthogonal to this feature. this test case seems adequate for now; if there is bug or regression, we will add necessary cases.